sycl/doc/extensions/RelaxStdLayout/SYCL_INTEL_relax_standard_layout.asciidoc

```diff
@@ -0,0 +1,160 @@
+
+// This is necessary for asciidoc, but not for asciidoctor
+:cpp: C++
+
+== Introduction
+IMPORTANT: This specification is a draft.
+
+NOTE: Khronos(R) is a registered trademark and SYCL(TM) and SPIR(TM) are trademarks of The Khronos Group Inc.  OpenCL(TM) is a trademark of Apple Inc. used by permission by Khronos.
+
+NOTE: This document is better viewed when rendered as html with asciidoctor.  GitHub does not render image icons.
+
+This document describes an extension that removes the requirement for data accessible on a device to be standard layout.  The trivially copyable requirement remains unchanged.
+
+== Name Strings
+
++SYCL_INTEL_relax_standard_layout+
+
+== Notice
+
+Copyright (c) 2020 Intel Corporation.  All rights reserved.
+
+== Status
+
+Working Draft
+
+This is a preview extension specification, intended to provide early access to a feature for review and community feedback. When the feature matures, this specification may be released as a formal extension.
+
+Because the interfaces defined by this specification are not final and are subject to change they are not intended to be used by shipping software products.
+
+== Version
+
+Built On: {docdate} +
+Revision: 1
+
+== Contact
+Michael Kinsner, Intel (michael 'dot' kinsner 'at' intel 'dot' com)
+
+== Dependencies
+
+This extension is written against the SYCL 1.2.1 specification, Revision 6.
+
+== Overview
+
+SYCL 1.2.1 required data stored into a buffer or passed as a kernel argument to be standard layout.  This is in addition to the data also being trivially copyable.  This extension relaxes the standard layout requirement while leaving the trivially copyable requirement intact.
+
+Standard layout does not guarantee ABI compatibility across devices or between a device and the host, but can help.  To avoid introducing incompatible data layouts, this extension also requires device compilers to validate that their layout of data is compatible with that used by the host.  There are multiple implementation approaches that can achieve this, although they are beyond the scope of this extension which describes the behavior only.
+
+== Modifications of SYCL 1.2.1 Specification
+
+=== Modify Sentence in Section 3.10 (Language restrictions in kernels)
+
+==== From:
+
+Sharing data structures between host and device code imposes certain restrictions, such as use of only user defined classes that are {cpp}11 standard layout classes for the data structures, classes that are {cpp}11 trivially copyable classes for the data structures, and in general, no pointers initialized for the host can be used on the device.
+
+==== To:
+
+Sharing data structures between host and device code imposes certain restrictions, such as use of only user defined classes that are {cpp}11 trivially copyable classes for the data structures, classes that are {cpp}11 trivially copyable classes for the data structures, and in general, no pointers initialized for the host can be used on the device.
```

Comment:
It would clearer with smaller lines.
For example to see that you can remove the repetition ", classes that are {cpp}11 trivially copyable classes for the data structures,"

Reply:
Fixed the bug, and will consider wrapping lines :)